spot_ros2_control/hardware/src/spot_hardware_interface.cpp

```diff
@@ -373,6 +414,128 @@ void SpotHardware::stop_state_stream() {
+
+  bosdyn::api::RobotCommand joint_command = ::bosdyn::client::JointCommand();
+  auto joint_res = command_client_->RobotCommand(joint_command);
+  if (!joint_res.status) {
+    RCLCPP_ERROR(rclcpp::get_logger("SpotHardware"), "Failed to activate joint control mode");
+    RCLCPP_ERROR(rclcpp::get_logger("SpotHardware"), "Message: %s", joint_res.status.DebugString().c_str());
+    return false;
+  }
+
+  auto robot_command_stream_resp = robot_->EnsureServiceClient<::bosdyn::client::RobotCommandStreamingClient>();
+  if (!robot_command_stream_resp) {
+    RCLCPP_ERROR(rclcpp::get_logger("SpotHardware"), "Could not create robot command streaming client");
+    return false;
+  }
+  command_stream_service_ = robot_command_stream_resp.response;
+
+  RCLCPP_INFO(rclcpp::get_logger("SpotHardware"), "Robot Command Streaming Client successfully created!");
+
+  // Fill in the parts of the joint streaming command request that are constant.
+  auto* joint_cmd = joint_request_.mutable_joint_command();
+  // Gain values https://github.com/boston-dynamics/spot-cpp-sdk/blob/master/cpp/examples/joint_control/constants.hpp
+  // NOTE: these should be different depending on number of joints the robot has (arm or none)
+  // Right now this is just temporary to see if we can get the commands accepted by robot
+  // This should be handled via a parameter in the future.
+  std::vector<float> kp;
+  std::vector<float> kd;
+
+  // Assign k values depending on if the robot has an arm or not
+  switch (njoints_) {
+    case spot_ros2_control::kNjointsArm:
+      kp = spot_ros2_control::arm_kp;
+      kd = spot_ros2_control::arm_kd;
+      break;
+    case spot_ros2_control::kNjointsNoArm:
+      kp = spot_ros2_control::no_arm_kp;
+      kd = spot_ros2_control::no_arm_kd;
+      break;
+    default:
+      RCLCPP_ERROR(rclcpp::get_logger("SpotHardware"), "WRONG # OF JOINTS");
+      return false;
+  }
+
+  joint_cmd->mutable_gains()->mutable_k_q_p()->Assign(kp.begin(), kp.end());
+  joint_cmd->mutable_gains()->mutable_k_qd_p()->Assign(kd.begin(), kd.end());
+
+  // Let it extrapolate the command a little
+  joint_cmd->mutable_extrapolation_duration()->CopyFrom(
+      google::protobuf::util::TimeUtil::NanosecondsToDuration(5 * 1e6));
+
+  // WITHOUT THIS NO COMMANDS WILL BE ACCEPTED!!!!
+  ::bosdyn::client::SetRequestHeader("SpotHardware", &joint_request_);
+
+  command_stream_started_ = true;
+  return true;
+}
+
+void SpotHardware::stop_command_stream() {
+  if (!command_stream_started_) {
+    RCLCPP_INFO(rclcpp::get_logger("SpotHardware"), "Command stream already stopped");
+    return;
+  }
+  RCLCPP_INFO(rclcpp::get_logger("SpotHardware"), "Stopping Command Stream");
+  command_stream_started_ = false;
+}
+
+void SpotHardware::send_command(const JointStates& joint_commands) {
+  const std::vector<float> position = joint_commands.position;
+  const std::vector<float> velocity = joint_commands.velocity;
+  const std::vector<float> load = joint_commands.load;
```

Comment:
@tcappellari-bdai nit: use const refs instead of copies
```suggestion
  const std::vector<float>& position = joint_commands.position;
  const std::vector<float>& velocity = joint_commands.velocity;
  const std::vector<float>& load = joint_commands.load;
```